src/hooks/useCartUpdateApi.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useRef } from 'react';
+
+const useCartUpdateApi = (productId: number, value: number, setValue: (value: number) => void) => {
+  const prevValue = useRef(value);
+
+  const resetValueOnError = (promise: Promise<Response>, properResStatus: number) => {
+    promise
+      .then((response) => {
+        if (response.status !== properResStatus) setValue(prevValue.current);
+        else prevValue.current = value;
+      })
+      .catch(() => {
+        setValue(prevValue.current);
+      });
```

Comment:
비동기문으로 바꾸는게 좋지 않을까요?

Reply:
```suggestion
    const response = await promise;

    try {
      if (response.status !== properResStatus) throw new Error('Response failed');
      prevValue.current = value;
    } catch {
      setValue(prevValue.current);
    }
```

이런 식으로 바꿔 쓰는 게 더 깔끔해 보이긴 하네요!

Reply:
```suggestion
    try {
      const response = await promise;

      if (response.status !== properResStatus) throw new Error('Response failed');
      prevValue.current = value;
    } catch {
      setValue(prevValue.current);
    }
```
이런 식으로 바꿔 쓰는 게 더 깔끔해 보이긴 하네요!